### PR TITLE
Docs: Improve QUnit.config information and split into separate pages

### PR DESCRIPTION
**Pull request by Krinkle**
* Add "version added" to all configuration options, based on History.md or git-blame.

* Add a blue "info" panel to highlight when an option is also also available as CLI option and/or in the HTML Reporter interface.

* Add various examples and rewrite some descriptions.

* Rename the "Configuration tools" category to "Extension interface", and keep all entries except QUnit.config.

* Split the QUnit.config entries into separate pages in the repurposed "Configuration" category.

  Since this new category is quite large, I decided to list only the "QUnit.config" parent in the sidebar when viewing pages outside that hierarchy.

* Maintain redirects for URLs of renamed pages.

### CODE REVIEW

**Comment by Krinkle** [comment]
@smcclure15 Would you mind proof-reading this? (Also feedback welcome on the overall approach!)

File: docs/config/current.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+</tr>
+</table>
+
+This property is not actually a configuration option, but is exposed under `QUnit.config` for use by plugins and other integrations. This offers access to QUnit's internal `Test` object at runtime.
+
+Internals may change without notice. When possible, use [QUnit.on](../callbacks/QUnit.on.md) or [other callbacks](../callbacks/index.md) instead.
+
+### Example
+
+Access `QUnit.config.current.testName` to observe the currently runing test's name.
```
Comment (smcclure15) [inline docs/config/current.md:28]: typo: running

File: docs/config/failOnZeroTests.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  <th>type</th>
+  <td markdown="span">`boolean`</td>
+</tr>
+<tr>
+  <th>default</th>
+  <td markdown="span">`true`</td>
+</tr>
+</table>
+
+By default, it is considered an error if no tests were loaded, or if no tests matched the current filter. Turning this option off means an empty test run will result in a success instead.
```
Comment (smcclure15) [inline docs/config/failOnZeroTests.md:25]: "Turning this option off" seems a little odd; maybe "Setting this option to `true`"...

File: docs/config/module.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+</tr>
+<tr>
+  <th>default</th>
+  <td markdown="span">`undefined`</td>
+</tr>
+</table>
+
+<p class="note" markdown="1">This option can also be set by URL query parameter.</p>
+
+When specify, only a single module will be run if its name is a complete case-insensitive match. If no module name matches, then no tests will be run.
```
Comment (smcclure15) [inline docs/config/module.md:27]: Typo? When specified...

File: docs/config/noglobals.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  <th>type</th>
+  <td markdown="span">`boolean`</td>
+</tr>
+<tr>
+  <th>default</th>
+  <td markdown="span">`false`</td>
+</tr>
+</table>
+
+Use this option to let QUnit track of which global variables and properties exist on the global object (e.g. `window` in browsers). When new global properties are found, they will result in test failures to you make sure your application and your tests are not leaking any state.
```
Comment (smcclure15) [inline docs/config/noglobals.md:25]: Typo: to let QUnit >keep< track of?

File: docs/config/testTimeout.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  <th>type</th>
+  <td markdown="span">`number` or `undefined`</td>
+</tr>
+<tr>
+  <th>default</th>
+  <td markdown="span">`undefined`</td>
+</tr>
+</table>
+
+This can be overridden on a per-test basis via [assert.timeout()](../assert/timeout.md). If you don't have per-test overrides, it is recommended to set this to a relatively high value (e.g. `3000` for 30 seconds) to avoid intermittent test failures from unrelated delays one might in a browser or CI service.
```
Comment (smcclure15) [inline docs/config/testTimeout.md:25]: Typo: `30000` for 30 seconds (off by a 0)

**Review by smcclure15 — APPROVED** [review]
I really like this breakout; just this past week I was sharing links of the main config page, looking for anchors or something to reference individual fields.

I had some trouble generating locally (first time with Ruby, and my Mac was having trouble finding some headers), but I did spot a few typos. I can review more if I can get that site generation working, or just spot it in production - I don't think there's too much risk.

File: docs/config/failOnZeroTests.md (hunk 2 above)
Comment (Krinkle) [reply docs/config/failOnZeroTests.md:25]: Agreed. Done!

**Pull request merged**